### PR TITLE
Numeric ID importer: support p-refinement

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

When importing data from a domain that differs only by p-refinement, the data will be interpolated only within the matching elements. This eliminates errors from interpolating boundary points in the wrong neighbor. Important for PB&J.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/IO/Importers/Actions/ReadVolumeData.hpp [hunk 1]
```diff
@@ -153,49 +155,34 @@ tuples::tagged_tuple_from_typelist<FieldTagsList> extract_element_data(
...
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords) {
+  const auto logical_coords = logical_coordinates(mesh);
+  ElementMap<Dim, Frame::Inertial> element_map{
+      element_id, domain.blocks()[element_id.block_id()]};
+  const auto mapped_inertial_coords =
+      element_map(logical_coords, time, functions_of_time);
+  if (not equal_within_roundoff(mapped_inertial_coords, inertial_coords)) {
+    ERROR_NO_TRACE(
+        "The source and target domain don't match on grid "
+        << element_id
```
Comment (knelli2) [inline src/IO/Importers/Actions/ReadVolumeData.hpp]: `"don't match in block " << element_id.block_id() << " within element " << element id`

File: src/IO/Importers/Actions/ReadVolumeData.hpp [hunk 2]
```diff
@@ -533,45 +545,57 @@ struct ReadAllVolumeDataAndDistribute {
...
-          source_domain_functions_of_time = deserialize<std::unordered_map<
-              std::string,
-              std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>>(
-              serialized_functions_of_time->data());
+      } else {
+        if (elements_match) {
+          Parallel::printf(
+              "WARNING: No serialized domain found in file. "
+              "Verification that elements in the source and target domain "
+              "match will be skipped.\n");
```
Comment (knelli2) [inline src/IO/Importers/Actions/ReadVolumeData.hpp]: I'm not sure we should allow this. Is there any scenario where somebody has an h5 file with volume data in the exact configuration of domain, but without the serialized domain?

File: src/IO/Importers/Tags.hpp [hunk 3]
```diff
@@ -68,15 +69,19 @@ struct ObservationValueEpsilon {
 /*!
  * \brief Toggle interpolation of numeric data to the target domain
  */
-struct EnableInterpolation {
-  static std::string name() { return "Interpolate"; }
+struct ElementsMatch {
```
Comment (knelli2) [inline src/IO/Importers/Tags.hpp]: This name is a bit obscure and I'm not sure users will have a sense of what this option does just from the name. A good name doesn't immediately stand out to me (and I see why you removed the word "interpolate" from the name since we can still do interpolation for matching grids). Here are some ideas but I'm open to more
- `InterpolateAcrossBlocks{Elements}` (or the opposite `InterpolateWithinElements`)
- `IdenticalVolumeFileDomain` ( `IdenticalH5Domain`, `IdenticalSerializedDomain`)
- `IdenticalDomains` ( `IdenticalDomainStructure`)

File: tests/InputFiles/GeneralizedHarmonic/KerrSchild.yaml [hunk 4]
```diff
@@ -60,15 +60,6 @@ Amr:
   Verbosity: Quiet
 
 PhaseChangeAndTriggers:
-  - Trigger:
```
Comment (knelli2) [inline tests/InputFiles/GeneralizedHarmonic/KerrSchild.yaml]: Why remove this? If you want to remove it, do so in a separate commit since it's unrelated to volume data reading

File: tests/Unit/IO/Importers/Test_VolumeDataReaderActions.cpp [hunk 5]
```diff
@@ -286,16 +302,18 @@ void test_actions(const std::variant<double, importers::ObservationSelector>&
...
+  CHECK_THROWS_WITH(test_actions<false>(0., false, false, true),
                     Catch::Matchers::ContainsSubstring("not supported"));
 
   for (const bool subcell_is_active : {false, true}) {
-    test_actions<true>(0., subcell_is_active);
-    test_actions<true>(importers::ObservationSelector::First,
+    test_actions<true>(0., false, subcell_is_active);
+    test_actions<true>(importers::ObservationSelector::First, false,
+                       subcell_is_active);
+    test_actions<true>(importers::ObservationSelector::Last, false,
```
Comment (knelli2) [inline tests/Unit/IO/Importers/Test_VolumeDataReaderActions.cpp:318]: If this feature isn't supported for subcell or FD grids, does it error in a clear way? Meaning do we print a message somewhere like "this isn't supported for FD grids" or will it error in an obscure place like when we compare the inertial coordinates?

**Review by knelli2 — CHANGES_REQUESTED** [review]

File: src/IO/Importers/Actions/ReadVolumeData.hpp (hunk 2 above)
Comment (nilsvu) [reply src/IO/Importers/Actions/ReadVolumeData.hpp]: Yes if they generate the file by providing data at given coordinates. It's important not to babysit users too much, i.e. allow the "I know what I'm doing is correct, just let me do it" scenario. Serializing a domain into the H5 file just to enable a check is too strict.

File: src/IO/Importers/Actions/ReadVolumeData.hpp (hunk 1 above)
Comment (nilsvu) [reply src/IO/Importers/Actions/ReadVolumeData.hpp]: The block ID is already printed as part of the element ID

File: tests/InputFiles/GeneralizedHarmonic/KerrSchild.yaml (hunk 4 above)
Comment (nilsvu) [reply tests/InputFiles/GeneralizedHarmonic/KerrSchild.yaml]: Oh this was just to run the input file without segments, dropped this change

File: src/IO/Importers/Tags.hpp (hunk 3 above)
Comment (nilsvu) [reply src/IO/Importers/Tags.hpp]: changed to `ElementsAreIdentical`

**Review by knelli2 — DISMISSED** [review]
Looks good. Squash in the clang-tidy fix as well

**Review by knelli2 — APPROVED** [review]
